python/ray/dag/tests/experimental/test_accelerated_dag.py

```diff
@@ -418,6 +418,94 @@ def test_actor_method_bind_same_input_attr(ray_start_regular):
     compiled_dag.teardown()
 
 
+def test_actor_method_bind_diff_input_attr_1(ray_start_regular):
+    actor = Actor.remote(0)
+    with InputNode() as inp:
+        # Two class methods are bound to two different input
+        # attribute nodes.
+        output1 = actor.inc.bind(inp[0])
+        output2 = actor.inc.bind(inp[1])
+        dag = MultiOutputNode([output1, output2])
+    compiled_dag = dag.experimental_compile()
+    ref = compiled_dag.execute(0, 1)
+    assert ray.get(ref) == [0, 1]
+
+    ref = compiled_dag.execute(1, 2)
+    assert ray.get(ref) == [2, 4]
+
+    compiled_dag.teardown()
+
+
+def test_actor_method_bind_diff_input_attr_2(ray_start_regular):
+    actor = Actor.remote(0)
+    with InputNode() as inp:
+        # Three class methods are bound to two different input
+        # attribute nodes. Two methods are bound to the same input
+        # attribute node.
+        output1 = actor.inc.bind(inp[0])
+        output2 = actor.inc.bind(inp[0])
+        output3 = actor.inc.bind(inp[1])
+        dag = MultiOutputNode([output1, output2, output3])
+    compiled_dag = dag.experimental_compile()
+    ref = compiled_dag.execute(0, 1)
+    assert ray.get(ref) == [0, 0, 1]
+
+    ref = compiled_dag.execute(1, 2)
+    assert ray.get(ref) == [2, 3, 5]
+
+    compiled_dag.teardown()
+
+
+def test_actor_method_bind_diff_input_attr_3(ray_start_regular):
+    actor = Actor.remote(0)
+    with InputNode() as inp:
+        # A single class method is bound to two different input
+        # attribute nodes.
+        dag = actor.inc_two.bind(inp[0], inp[1])
+    compiled_dag = dag.experimental_compile()
+    ref = compiled_dag.execute(0, 1)
+    assert ray.get(ref) == 1
+
+    ref = compiled_dag.execute(1, 2)
+    assert ray.get(ref) == 4
+
+    compiled_dag.teardown()
+
+
+def test_actor_method_bind_diff_input_attr_4(ray_start_regular):
+    actor = Actor.remote(0)
+    with InputNode() as inp:
+        output1 = actor.inc_two.bind(inp[0], inp[1])
+        output2 = actor.inc.bind(inp[2])
```

Comment:
Can you add a test that there are overlaps on input attributes between two calls?

```
        output1 = actor.inc_two.bind(inp[0], inp[1])
        output2 = actor.inc_two.bind(inp[2], inp[0])
```

Reply:
updated: https://github.com/ray-project/ray/pull/47489/commits/acdca29ff3c95622bf47b59db01ab114baa91eed